src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -394,20 +372,16 @@ private boolean validWriterConnection() {
             || isWriterHostIndex(this.currentHostIndex);
   }
 
-  private void attemptConnectionUsingCachedTopology() throws SQLException {
+  private void attemptConnectionUsingCachedTopology(String connUrl) throws SQLException {
     List<HostInfo> cachedHosts = topologyService.getCachedTopology();
     if (Util.isNullOrEmpty(cachedHosts)) {
-      if (this.gatherPerfMetricsSetting) {
```

Comment:
Why don't we check for `gatherPerfMetricsSetting` anymore?

Reply:
Checking inside container & having it update/check if it's been changed because it's runtime-modifiable